components/channel_view/channel_view.jsx

```diff
@@ -145,10 +152,19 @@ export default class ChannelView extends React.PureComponent {
                 <ChannelHeader
                     channelId={this.props.channelId}
                 />
-                <DeferredPostView
-                    channelId={this.props.channelId}
-                />
-                {createPost}
+                { this.props.channelLoading ? (
+                    <LoadingScreen
```

Comment:
Yes, I will create a ticket for this. 
Note: This might actually show two loaders. one for channel loading and one for post-loading so, the closer they are the better so don't flicker it too much